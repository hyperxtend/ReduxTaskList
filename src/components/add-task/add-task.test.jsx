import React from 'react'
import { shallow, mount } from 'enzyme'
import AddTask, { handleAddTodo } from './add-task'
import store from "../../redux/store";
 


describe('<AddTask/>', () => {
    const component = shallow(<AddTask store={store}/>);
    const submitMock = jest.fn();
    let comp;
    
    it('should render', () => {
        expect(component.exists()).toBe(true);
    });
    it('checks empty input', () => {
        expect(component.find('.inputTask').length).toEqual(0);
    });
    it('Should call the submitTodo function when clicked', () => {
        comp = mount(<AddTask store={store} handleAddTodo={submitMock} />);
        expect(submitMock.mock.calls.length).toEqual(0);
        comp.find('form').simulate('submit');
        expect(submitMock.mock.calls.length).toEqual();
        // wrapper.find('form').at(0).simulate('submit');
        // expect(onSubmitSpy).toHaveBeenCalled();
      });
});
