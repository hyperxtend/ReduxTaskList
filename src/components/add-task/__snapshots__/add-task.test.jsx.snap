// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTask/> Should call the submitTodo function when clicked 1`] = `ShallowWrapper {}`;
